src/Components/SingleRoom/SingleRoom.tsx

```diff
@@ -1,29 +1,106 @@
 import "./SingleRoom.css";
 import {Player} from "../Player/Player";
 import {Card} from "../Card/Card";
+import {useEffect, useState} from "react";
+import AwesomeButtonStyles from "../Buttons/styles/Themes/theme-c137/styles.module.scss";
+import {AwesomeButton} from "react-awesome-button";
+import {apiUrl} from "../../config/api";
+import {CardArrayInterface, changeCardNaming} from "../utils/changeCardNaming";
+import {io} from "socket.io-client";
+
+const socket = io(`${apiUrl}`);
 
 export const SingleRoom = () => {
-    const playerName = "John"
+    const playerName = "John";
     const cards = ['2c', '3c', '4c', '5c', '6c', '7c', '8c', '9c', 'Tc', 'Jc', 'Qc', 'Kc', 'Ac',
         '2d', '3d', '4d', '5d', '6d', '7d', '8d', '9d', 'Td', 'Jd', 'Qd', 'Kd', 'Ad',
         '2h', '3h', '4h', '5h', '6h', '7h', '8h', '9h', 'Th', 'Jh', 'Qh', 'Kh', 'Ah',
-        '2s', '3s', '4s', '5s', '6s', '7s', '8s', '9s', 'Ts', 'Js', 'Qs', 'Ks', 'As',]
+        '2s', '3s', '4s', '5s', '6s', '7s', '8s', '9s', 'Ts', 'Js', 'Qs', 'Ks', 'As',];
 
+    const [playersCards, setPlayersCards] = useState<string[][]>([['', '']]);
+    const [commonCards, setCommonCards] = useState<string[]>(['', '', '', '', '']);
     const cardHeight = 120;
+    useEffect(() => {
+        socket.emit('turn', {}, (response: CardArrayInterface[]) => {
+            console.log(response)
+
+        });
+
+        return () => {
+            socket.off('turn');
+        }
+    },[]);
```

Comment:
please export to separate hook

```suggestion
export function useFetchCommonCardsFromServer() {
    useEffect(() => {
        socket.emit('turn', {}, (response: CardArrayInterface[]) => {
            console.log(response)
        });

        return () => {
            socket.off('turn');
        }
    },[]);
 }
```

```
export const SingleRoom = () => {
   //. ...
   useFetchCommonCardsFromServer();
```